frontend/integration-tests/tests/storage.scenario.ts

```diff
@@ -0,0 +1,63 @@
+import { $, $$, browser, ExpectedConditions as until, by, element } from 'protractor';
+import { testName, checkLogs, checkErrors } from '../protractor.conf';
+import * as crudView from '../views/crud.view';
+import * as storageView from '../views/storage.view';
+import { execSync } from 'child_process';
+
+describe('Add storage is applicable for all workloads', () => {
+
+  const k8sWorkloads = ['replicationcontrollers', 'daemonsets', 'deployments', 'replicasets', 'statefulsets'];
+  const openshiftWorkloads = ['deploymentconfigs'];
+  const resourceObjs = browser.params.openshift === 'true' ? k8sWorkloads.concat(openshiftWorkloads) : k8sWorkloads;
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  afterAll(() => {
+    resourceObjs.forEach((resourceType) => {
+      try {
+        execSync(`kubectl delete --cascade ${resourceType} example -n ${testName}`);
+      } catch (error) {
+        console.error(`Failed to delete ${resourceType} example:\n${error}`);
+      }
+    });
+  });
+
+  resourceObjs.forEach((resourceType) => {
+    describe(resourceType, () => {
+      const pvcName = `${resourceType}-pvc`;
+      const pvcSize = '1';
+      const mountPath = '/data';
+      it(`create a ${resourceType} resource`, async() => {
+        await crudView.createNamespacedResourceWithDefaultYaml(resourceType);
+        expect(crudView.errorMessage.isPresent()).toBe(false);
+      });
+      it(`add storage to ${resourceType}`, async() => {
+        if (resourceType === 'deploymentconfigs' || resourceType === 'deployments') {
+          const activeStatus = element(by.cssContainingText('.co-icon-and-text', 'Up to date'));
+          if (!activeStatus.isPresent()) {
+            await browser.sleep(20000);
+          }
+          const availableCondition = $$('[data-test-condition="Available"] div').get(1);
```

Comment:
Add a `data-test-` attribute on the exact column you want instead of just on the row.

Reply:
Do we need to wait for the deployment to be available before continuing with the test? I believe we use patch, so you shouldn't see errors even if it's still rolling out.

Reply:
It often failed in ci flow with this error, haven't you met with this error?
![Screenshot from 2019-07-03 10-40-14](https://user-images.githubusercontent.com/7313189/60567660-3e2be080-9d9d-11e9-9f04-e5da592c6f6a.png)